pkg/util/aws/shared_credentials_file.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SharedCredentialsFileFromSecret returns a location (path) to the shared credentials file that was created
+// using the provided secret.
+// Consumers can configure the aws Session using that file to use the credentials like,
+//
+// sharedCredentialsFile, err := SharedCredentialsFileFromSecret(secret)
+// if err != nil {
+// 	// handle error
+// }
+// options := session.Options{
+// 	SharedConfigState: session.SharedConfigEnable,
+// 	SharedConfigFiles: []string{sharedCredentialsFile},
+// }
+// sess := session.Must(session.NewSessionWithOptions(options))
+//
+func SharedCredentialsFileFromSecret(secret *corev1.Secret) (string, error) {
+	var data []byte
+	switch {
+	case len(secret.Data["credentials"]) > 0:
+		data = secret.Data["credentials"]
+	case len(secret.Data["aws_access_key_id"]) > 0 && len(secret.Data["aws_secret_access_key"]) > 0:
+		data = newConfigForStaticCreds(
+			string(secret.Data["aws_access_key_id"]),
+			string(secret.Data["aws_secret_access_key"]),
+		)
+
+	default:
+		return "", errors.New("invalid secret for aws credentials")
+	}
+
+	f, err := ioutil.TempFile("", "aws-shared-credentials")
```

Comment:
What cleans up old temporary files when the token changes?

Reply:
see https://github.com/openshift/cluster-ingress-operator/pull/485/files#diff-cfa88f3094d390a95d1d9b230dadae57aefa04e78041f7697f941ca110f7bca4R450-R452 for when cleanup happens.